Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/PatientDiscovery201305Processor.java

```diff
@@ -336,128 +350,142 @@ protected void createPatientCorrelation(PRPAIN201306UV02 queryResult, AssertionT
+                    LOG.error(
+                        "Remote patient identifiers are null. Could not correlate the patient identifiers.");
                 }
+            } else {
+                LOG.error(
+                    "Request (PRPAIN201301UV02) or remote patient identifiers are null. Could not correlate the patient identifiers.");
             }
         } else {
-            LOG.error("Null parameter passed to createPatientCorrelation method, no correlation created");
+            LOG.error("Local patient identifiers are null. Could not correlate the patient identifiers.");
         }
     }
 
+    private static boolean checkPatientInRegistrationEvent(PRPAIN201301UV02 request) {
+        boolean result = false;
+
+        if(request.getControlActProcess().getSubject().get(0).getRegistrationEvent() != null){
+
+            result = request.getControlActProcess().getSubject().get(0).getRegistrationEvent()
+                .getSubject1() != null
+                && request.getControlActProcess().getSubject().get(0).getRegistrationEvent().getSubject1()
+                .getPatient() != null
+                && NullChecker.isNotNullish(request.getControlActProcess().getSubject().get(0)
+                    .getRegistrationEvent().getSubject1().getPatient().getId());
+        }
+        return result;
+    }
     private II providedPatientId(PRPAIN201305UV02 request) {
         II patId = null;
         String aaId;
 
-        try {
-            if (request != null && request.getControlActProcess() != null) {
-                aaId = extractAAOID(request);
-
-                if (NullChecker.isNotNullish(aaId) && requestHasLivingId(request)) {
-                    for (PRPAMT201306UV02LivingSubjectId livingSubId : request.getControlActProcess()
-                        .getQueryByParameter().getValue().getParameterList().getLivingSubjectId()) {
-                        for (II id : livingSubId.getValue()) {
-                            if (id != null && NullChecker.isNotNullish(id.getRoot())
-                                && NullChecker.isNotNullish(id.getExtension())
-                                && aaId.contentEquals(id.getRoot())) {
-                                patId = new II();
-                                patId.setRoot(id.getRoot());
-                                patId.setExtension(id.getExtension());
-
-                                // break out of inner loop
-                                break;
-                            }
-                        }
-
-                        // If the patient id was found then break out of outer loop
-                        if (patId != null) {
+        if (request != null && request.getControlActProcess() != null) {
+
+            aaId = extractAAOID(request);
+
+            if (NullChecker.isNotNullish(aaId) && request.getControlActProcess().getQueryByParameter() != null
```

Comment:
Seem like you introduce long null check instead of calling requestHasLivingId(request)? any particular reasons?

Reply:
I guess my changes were overridden during merge conflict.